packages/babel-parser/src/plugins/typescript/index.js

```diff
@@ -2625,25 +2617,15 @@ export default (superClass: Class<Parser>): Class<Parser> =>
           // e.g. `const f = (foo: number = 0) => foo;`
           return;
         case "TSParameterProperty":
-          this.checkLVal(
-            expr.parameter,
-            bindingType,
-            checkClashes,
-            "parameter property",
-          );
+          this.checkLVal(expr.parameter, "parameter property", ...args);
           return;
         case "TSAsExpression":
         case "TSNonNullExpression":
         case "TSTypeAssertion":
-          this.checkLVal(
-            expr.expression,
-            bindingType,
-            checkClashes,
-            contextDescription,
-          );
+          this.checkLVal(expr.expression, contextDescription, ...args);
           return;
         default:
-          super.checkLVal(expr, bindingType, checkClashes, contextDescription);
+          super.checkLVal(expr, contextDescription, ...args);
```

Comment:
Passing through all the rest parameters here causes typescript-parser tests failed: https://travis-ci.com/github/babel/babel/jobs/433402908#L842

Ironically this is a bug fix, let's see one of the failing tests

https://github.com/microsoft/TypeScript/blob/b5b0437a86661c8d7bc76c5860c07305df17899c/tests/cases/compiler/letInConstDeclarations_ES5.ts#L1-L8

It should fail but TS compiler is so loose that it allows such cases, just to be thrown later by TS type checking diagnostics. I will add them to the `allowlist`.